src/library/PiaFormat.ts

```diff
@@ -0,0 +1,890 @@
+          ),
+        }
+      : {};
+    let line8Data = line8
+      ? {
+          futureProjectionStubString: piaSubstr(
+            line8,
+            this.fieldFormats.futureProjectionStubString.startChar,
+            this.fieldFormats.futureProjectionStubString.endChar
+          ),
+        }
+      : {};
+    return { ...line7Data, ...line8Data };
+  }
+})();
+
+const PIA_SERIALIZERS: PIASerializer[] = [
+  basicInfoSerializer,
+  benefitSerializer,
+  oasdiEarningsSerializer,
+  monthlyNoncoveredPensionSerializer,
+  nameOfWorkerSerializer,
+  oldQuartersOfCoverageSerializer,
+  wageBaseSerializer,
+  earningsProjectionStubSerializer, //remove me as soon as implemented in oasdiEarningSerializer
+  // disabilityDatesSerializer,
+  //…
+];
+
+/**
+ * Given a list of lines in the AnyPIA format, return a map from
+ * line number to the line string itself.
+ */
+function createLineMap(lines: string[]): PIALineMap {
+  return lines.reduce((lineMap, line) => {
+    const lineNum = parseInt(line.slice(0, 2), 10);
+    return Object.assign(lineMap, { [lineNum]: line });
+  }, {});
+}
+
+function deserializePIAData(lines: string[]): PIAData {
+  const lineMap = createLineMap(lines);
+  const deserializedData = PIA_SERIALIZERS.reduce(
+    (data, serializer) => Object.assign(data, serializer.deserialize(lineMap)),
+    {}
+  );
+
+  return deserializedData;
+}
+
+function serializePIAData(data: PIAData): PIALineMap {
+  const lines = PIA_SERIALIZERS.reduce(
+    (lineMap, serializer) => Object.assign(lineMap, serializer.serialize(data)),
+    {}
+  );
+  console.log(data);
+  return lines;
+}
+
+enum PIASex {
+  male = 0,
+  female = 1,
+}
+
+enum SSABenefitType {
+  old_age = 1,
+  survivor = 2,
+  disability = 3,
+}
+
+/* http://thadk.net/anypiamac-docs/html/Forms/type_of_earnings.html */
+enum PIATypeOfEarnings {
+  entered_earnings = 0,
+  maximum = 1,
+  high = 2,
+  average = 3,
+  low = 4,
+}
+
+enum PIATypeOfTaxes {
+  employee_taxes = 0,
+  self_employed_taxes = 1,
+}
+
+interface PIAData {
+  ssn?: string; //line1
+  birthDate?: PIADate;
+  sex?: PIASex;
+  dateOfDeath?: PIADate; //line2
+  typeOfBenefit?: SSABenefitType; //line 4
+  monthYearBenefit?: PIAMonthYear;
+  monthYearEntitlement?: PIAMonthYear;
+  firstEarningYearActual?: PIAYear; //line6
+  lastEarningYearActual?: PIAYear;
+  typeOfEarnings?: Map<PIAYear, PIATypeOfEarnings>; //line20
+  typeOfTaxes?: Map<PIAYear, PIATypeOfTaxes>; //line21
+  oasdiEarnings?: Map<PIAYear, PIAEarnings>; //line22-29
+  hiEarnings?: Map<PIAYear, PIAEarnings>;
+  monthlyNoncoveredPensionAmount?: PIAEarnings; //line12
+  monthYearEntitlementNoncoveredPension?: PIAMonthYear;
+  nameOfWorker?: string; //line16
+  oldQuartersOfCoverageStubString?: string;
+  wageBaseStubString?: string;
+  pastProjectionStubString?: string;
+  futureProjectionStubString?: string;
+  piaEverythingElse?: string;
+}
+
+export class PiaFormat {
+  piaAll: string;
```

Comment:
lets remember it keeps an all-inclusive string from the constructor as it stands: do we still want this?